src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3289,11 +3298,10 @@ void MacroAssembler::compiler_fast_unlock_object(Register oop, Register box, Reg
   Register displacedHeader = temp1;
   Register currentHeader = temp2;
   Register temp = temp1;
-  Register monitor = temp2;
 
   const int hdr_offset = oopDesc::mark_offset_in_bytes();
```

Comment:
Either use this aux. variable in both, lock and unlock, methods or in none. I prefer using the aux. variable.